core/src/main/java/io/lionweb/lioncore/java/serialization/JsonSerialization.java

```diff
@@ -529,8 +634,25 @@ private List<ClassifierInstance<?>> deserializeClassifierInstances(
         .forEach(
             node -> {
               populateClassifierInstance(
-                  node, serializedToInstanceMap.get(node), classifierInstanceResolver);
+                  node,
+                  serializedToInstanceMap.get(node),
+                  classifierInstanceResolver,
+                  sortingResult);
               ClassifierInstance<?> classifierInstance = serializedToInstanceMap.get(node);
+              if (unavailableParentPolicy == UnavailableNodePolicy.PROXY_NODES) {
+                // For real parents, the parent is not set directly, but it is set indirectly
```

Comment:
I am not sure I have a reasonable solution for that. We could set the children we know of in the proxy parent node, but then the proxy would have only a _partial_ list of children (just the ones that we are aware of, as we do not retrieve the data of the proxy node). So we could use the proxy nodes to add children in it only for the purpose of it triggering `setParent` on those children. I do not think this would make the situation better and I cannot think of other options

Reply:
I agree, doesn't sound better.